lib/charms/vault_k8s/v0/juju_facade.py

```diff
@@ -372,8 +373,18 @@ def get_relations(self, relation_name: str, relation_id: int | None = None) -> L
             if not relation_id
             else [self.get_relation_by_id(relation_name, relation_id)]
         )
+        if not relations:
+            logger.warning("No relations found for %s", relation_name)
         return relations
 
+    def get_active_relation_by_id(self, relation_name: str, relation_id: int) -> Relation | None:
```

Comment:
The name contradicts with the input, the name suggests we get it by id but we get it by name and id, i'd suggest replacing with `get_active_relation`.